### PR TITLE
WIP: OSSM-1211 Merge a locality setting from an imported service

### PR DESCRIPTION
**Pull request by yxun**
Signed-off-by: Yuanlin <yuanlin.xu@redhat.com>

**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by rcernich** [comment]
Is it possible to add a unit test that recreates the bug and verifies the fix?

**Comment by yxun** [comment]
This change should help set a locality when the controller defaultLocality is empty. See the following line:
https://github.com/maistra/istio/blob/maistra-2.1.2/pilot/pkg/serviceregistry/federation/controller.go#L255

In our 2.1 code, we ignore the locality spec value from the importConfig when c.defaultLocality is nil. 
I think we should not ignore that.

**Comment by rcernich** [comment]
Please add some unit tests.

File: pilot/pkg/serviceregistry/federation/controller.go [hunk 1]
```diff
@@ -136,6 +136,8 @@ func mergeLocality(locality *v1.ImportedServiceLocality, defaults *v1.ImportedSe
 		if locality == nil {
 			return nil
 		}
+		// When defaults is nil and locality is not nil
+		merged = *locality
 	} else if locality != nil {
 		merged = *locality
 	}
```
Comment (jewertow) [inline pilot/pkg/serviceregistry/federation/controller.go:143]: Couldn't this if-else condition be simplified to the following conditions?
```
if defaults == nil {
	defaults = &v1.ImportedServiceLocality{}
	if locality == nil {
		return nil
	}
}
if locality != nil {
	merged = *locality
}
```

**Comment by yxun** [comment]
comment added in OSSM-1211. This PR is not related to the OSSM-1211 issue so close this one now.

**Pull request closed without merging**